WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -205,6 +225,16 @@ public void Spend(SmartCoin spentCoin, SmartTransaction tx)
 				{
 					SpentCoinsByOutPoint.Add(spentCoin.Outpoint, spentCoin);
 				}
+
+				// Update transaction amount value.
+				if (TransactionAmountsByTxid.TryGetValue(tx.GetHash(), out Money? amount))
+				{
+					TransactionAmountsByTxid[tx.GetHash()] = amount - spentCoin.Amount;
+				}
+				else
+				{
+					TransactionAmountsByTxid[tx.GetHash()] = Money.Zero - spentCoin.Amount;
+				}
```

Comment:
Should we move this to its own function? we are using it twice. Same for other caches but unrelated.